File: data/EXP01/python/2452/2452.output_3.py
# @lc app=leetcode id=2452 lang=python3
class Solution:
    def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
        def generate_edit_words(word, edits):
            words = set()
            for i in range(len(word)):
                for letter in 'abcdefghijklmnopqrstuvwxyz':
                    new_word = word[:i] + letter + word[i+1:]
                    if edits == 2:
                        words.update(generate_edit_words(new_word, edits - 1))
                    else:
                        words.add(new_word)
            return words
        
        result = []
        for query in queries:
            edit_words_1 = generate_edit_words(query, 1)
            edit_words_2 = generate_edit_words(query, 2)
            
            if any(word in dictionary for word in edit_words_1) or any(word in dictionary for word in edit_words_2):
                result.append(query)
        
        return result

File: data/EXP01/golang_no_system_var/2452/2452.output_3.go
func twoEditWords(queries []string, dictionary []string) []string {
    result := []string{}
    
    for _, query := range queries {
        if containsWord(query, dictionary) || canTransform(query, dictionary) {
            result = append(result, query)
        }
    }
    
    return result
}

func containsWord(word string, dictionary []string) bool {
    for _, dictWord := range dictionary {
        if word == dictWord {
            return true
        }
    }
    return false
}

func canTransform(word string, dictionary []string) bool {
    for _, dictWord := range dictionary {
        if isOneEditAway(word, dictWord) || isTwoEditsAway(word, dictWord) {
            return true
        }
    }
    return false
}

func isOneEditAway(word1, word2 string) bool {
    diffCount := 0
    for i := range word1 {
        if word1[i] != word2[i] {
            diffCount++
            if diffCount > 1 {
                return false
            }
        }
    }
    return diffCount == 1
}

func isTwoEditsAway(word1, word2 string) bool {
    for i := range word1 {
        if word1[i] != word2[i] {
            editedWord := word1[:i] + word2[i:i+1] + word1[i+1:]
            if editedWord == word2 {
                return true
            }
            for j := range editedWord {
                if j != i && editedWord[j] != word2[j] {
                    editedWord = editedWord[:j] + word2[j:j+1] + editedWord[j+1:]
                    if editedWord == word2 {
                        return true
                    }
                }
            }
            return false
        }
    }
    return false
}
